lib/core/include/scipp/core/element/to_unit.h

```diff
@@ -40,7 +40,8 @@ constexpr auto round = [](const auto x) {
 
 constexpr auto to_unit = overloaded{
     arg_list<double, std::tuple<float, double>, std::tuple<int64_t, double>,
-             std::tuple<int32_t, double>, std::tuple<time_point, double>,
+             std::tuple<int64_t, int64_t>, std::tuple<int32_t, double>,
```

Comment:
I don't understand why you didn't add `tuple<int_32, int_32>`. What would be inconsistent?

Reply:
When converting, e.g., from seconds to nanoseconds, int32 would almost always be too small to represent the result. In all other cases we cast the output to the input type, if we were now to use, e.g., an int64 scale we would only benefit if we also *return* int64.

Reply:
> int32 would almost always be too small to represent the result.

For datetimes, yes. But the `int` overloads are used for durations.

> int64 scale we would only benefit if we also return int64.

But what about an int32 scale? Are errors in `double` always small enough to not affect int32?

Reply:
I would think so? int32 max is if O(1e9), and double machine precision is much higher?

Reply:
Sounds right